build-support/bin/mypy.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import argparse
+import subprocess
+from typing import List
+
+from common import die
+
+
+def main():
+  targets = create_parser().parse_args().targets
+  run_mypy(targets)
+
+
+def create_parser() -> argparse.ArgumentParser:
+  parser = argparse.ArgumentParser(description="Run MyPy with our config file.")
+  parser.add_argument(
+    "targets",
+    default=["build-support::", "src/python::", "tests/python::", "contrib::"],
+    nargs="*",
+    help="Pants targets, e.g. `tests::`.",
+  )
+  return parser
+
+
+def run_mypy(targets: List[str]) -> None:
+  command = [
+    "./pants",
+    "mypy",
+    "--mypy-mypy-version=0.701",
+    "--config-file=build-support/mypy/mypy.ini",
+  ]
+  try:
+    subprocess.run(command + targets, check=True)
+  except subprocess.CalledProcessError:
+    die("Please fix the above type errors and run again.")
```

Comment:
I'm not sure how valuable this would be to point to the config, especially since our config is almost entirely turning _off_ checks. Our config is very loose—unlike the setup repo.

Further, MyPy will output any rules violations and explain what went wrong. For example, let's pretend we have the rule `disallow_untyped_defs = True`, then run `./build-support/bin/mypy.py tests/python/pants_test/util:strutil`. We would get back:
```
08:28:34 00:02   [mypy]
08:28:34 00:02     [mypy]
08:28:34 00:02       [check]
src/python/pants/util/strutil.py:14:1: error: Function is missing a type annotation
src/python/pants/util/strutil.py:23:1: error: Function is missing a type annotation
...
tests/python/pants_test/util/test_strutil.py:16:3: error: Function is missing a type annotation
tests/python/pants_test/util/test_strutil.py:33:3: error: Function is missing a type annotation
...

FAILURE: mypy failed: code=1


               Waiting for background workers to finish.
08:28:36 00:04   [complete]
               FAILURE
Please fix the above type errors and run again.
```

Finally, we don't expect devs to modify `pants.ini` very much, so I think it may be more cognitive load than necessary.

Reply:
It feels like this error message should be incorporated into the task help instead? But yea, mentioning the (relative) path to the config file would be useful.